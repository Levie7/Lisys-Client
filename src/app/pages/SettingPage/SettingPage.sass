@import 'src/style/lisys/settings/settings.radius'
@import 'src/style/lisys/settings/settings.shadow'

.SettingCard
    border: none
    border-radius: $radius-large
    box-shadow: $shadow-card
    height: 100%
    max-width: 1366px
    min-height: 620px
    width: 100%
