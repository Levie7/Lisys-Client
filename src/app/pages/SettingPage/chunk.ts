import { PageLoading } from 'src/app/shell/PageLoading';

import { suspend } from 'src/core/chunk';

export const SettingPage = suspend({
    component: () => import(/* webpackChunkName: "SettingPage" */ './index'),
    renderWhileLoading: PageLoading,
});
