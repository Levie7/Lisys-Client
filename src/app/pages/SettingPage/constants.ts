export const tabData = [
    {
        category: 'company',
        description: { en: 'Information, Logo', id: 'Informasi, Logo' },
        icon: 'shop',
        key: '1',
        title: { en: 'Company', id: 'Perusahaan' },
        type: 'default',
    },
    {
        category: 'general',
        description: { en: 'Format Date & Time, Language', id: 'Format Tanggal & Waktu, Bahasa' },
        icon: 'global',
        key: '2',
        title: { en: 'General', id: 'Umum' },
        type: 'default',
    },
    {
        category: 'user_management',
        description: { en: 'User, Permission, Role', id: 'Pengguna, Izin, Peran' },
        icon: 'user',
        key: '3',
        title: { en: 'User Management', id: 'Managemen Pengguna' },
        type: 'section',
    },
];
