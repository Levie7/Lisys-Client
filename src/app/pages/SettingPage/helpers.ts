import { SettingLanguage } from 'src/core/api';

export function getLanguage(): SettingLanguage {
    return localStorage.getItem('language') === 'English' ? 'en' : 'id';
}
