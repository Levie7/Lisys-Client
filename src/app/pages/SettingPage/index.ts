export { SettingPage as default } from './SettingPage';
