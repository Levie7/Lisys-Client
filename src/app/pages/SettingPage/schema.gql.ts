import { gql, useQuery } from 'src/core/graphql';

const MOVIES = gql`
    {
        movies {
            mission_name
        }
    }
`;

export const useMovies = () => useQuery<{ movies: [{ mission_name: string }] }>(MOVIES);
