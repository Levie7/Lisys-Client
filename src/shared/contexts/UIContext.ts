import * as React from 'react';

export interface UIContextValue {
    isMobile: boolean;
}

const UIContext = React.createContext<UIContextValue>({
    isMobile: true,
});
export default UIContext;
export const useUIContext = () => React.useContext(UIContext);
