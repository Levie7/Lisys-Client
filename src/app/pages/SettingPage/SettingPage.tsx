import React from 'react';

import Company from 'src/modules/components/Settings/Company/Company';
import General from 'src/modules/components/Settings/General/General';
import { UserManagement } from 'src/modules/components/Settings/UserManagement';

import { Card } from 'src/shared/components/Card';
import { Icon } from 'src/shared/components/Icon';
import { Tabs, TabPositions } from 'src/shared/components/Tabs';
import { UIContext } from 'src/shared/contexts/UIContext';

import { tabData } from './constants';

export default function SettingPage() {
    function renderContent(tab: string) {
        switch (tab) {
            case 'Company':
                return <Company />;
            case 'General':
                return <General />;
            case 'User Management':
                return <UserManagement />;
            default:
                return null;
        }
    }

    function renderTabs(position: TabPositions) {
        return (
            <Tabs defaultActiveKey='2' tabPosition={position}>
                {tabData.map((tab) => (
                    <Tabs.TabPane
                        key={tab.key}
                        tab={
                            <div className='d-flex fd-row'>
                                <Icon className='d-flex fa-center' type={tab.icon} />
                                <div>
                                    <div className='fw-bold'>{tab.title}</div>
                                    <UIContext.Consumer>
                                        {({ isMobile }) =>
                                            !isMobile && (
                                                <div className='fg-gray-light'>
                                                    {tab.description}
                                                </div>
                                            )
                                        }
                                    </UIContext.Consumer>
                                </div>
                            </div>
                        }
                    >
                        {renderContent(tab.title)}
                    </Tabs.TabPane>
                ))}
            </Tabs>
        );
    }

    return (
        <div className='d-flex fj-center m-4'>
            <Card>
                <UIContext.Consumer>
                    {({ isMobile }) => (isMobile ? renderTabs('top') : renderTabs('left'))}
                </UIContext.Consumer>
            </Card>
        </div>
    );
}
