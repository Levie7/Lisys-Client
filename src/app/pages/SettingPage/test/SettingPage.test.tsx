import { shallow } from 'enzyme';
import React from 'react';

import UIContext, * as UIContextModule from 'src/shared/contexts/UIContext';

import SettingPage from '../SettingPage';

describe('SettingPage', () => {
    let context = { isMobile: false };
    jest.spyOn(UIContextModule, 'useUIContext').mockImplementation(() => context);
    let wrap = shallow(
        <UIContext.Provider value={context}>
            <SettingPage />
        </UIContext.Provider>,
        { context }
    );

    it('<SettingPage/> - should render SettingPage component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should render setting page component', () => {
        expect(wrap.find('SettingPage').exists()).toBeTruthy();
    });

    describe('when is not mobile', () => {
        beforeEach(() => {
            context = { isMobile: false };
            wrap.dive().setContext(context);
        });

        it('should render setting tabs with position left', () => {
            expect(
                wrap
                    .dive()
                    .find('Memo(SettingTabsPure)')
                    .exists()
            ).toBeTruthy();
            expect(
                wrap
                    .dive()
                    .find('Memo(SettingTabsPure)')
                    .props().position
            ).toEqual('left');
        });
    });

    describe('when is mobile', () => {
        beforeEach(() => {
            context = { isMobile: true };
            wrap.dive().setContext(context);
        });

        it('should render setting tabs with position top', () => {
            expect(
                wrap
                    .dive()
                    .find('Memo(SettingTabsPure)')
                    .exists()
            ).toBeTruthy();
            expect(
                wrap
                    .dive()
                    .find('Memo(SettingTabsPure)')
                    .props().position
            ).toEqual('top');
        });
    });
});
