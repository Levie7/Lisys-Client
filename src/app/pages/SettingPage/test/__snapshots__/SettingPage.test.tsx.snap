// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingPage <SettingPage/> - should render SettingPage component correctly 1`] = `ShallowWrapper {}`;
